gemfiles/rails_7_1.gemfile

```diff
@@ -0,0 +1,31 @@
+# This file was generated by Appraisal
+
+source "http://rubygems.org"
+
+gem "rake"
+gem "appraisal"
+gem "activerecord", "~> 7.1.0"
+
+group :development do
+  gem "github_changelog_generator", "1.9.0"
+end
+
+group :test do
+  gem "minitest", "~> 5.0"
+  gem "timecop"
+  gem "mocha"
+end
+
+group :sqlite do
+  gem "sqlite3", "~> 1.4"
+end
+
+group :postgresql do
+  gem "pg", "~> 1.2.0"
```

Comment:
I am not sure about this one. I run `bundle exec appraisal install` while  using ruby 3.1.5p252 (2024-04-23 revision 1945f8dc0e) [x86_64-linux] and I get  _gem "pg", "~> 1.2.0"_

However, on rails_7_0.gemfile, it uses 
`  gem "pg", "~> 1.3.0"`

```
Tried running it with other higher versions of ruby but always get this error
```
Installing pg 1.2.3 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
```

Reply:
I've found that you can use this line to help with that installation:

bundle config build.pg --with-cflags="-Wno-error=implicit-function-declaration"

However, the whole appraisal testing thing is a real pain. I was able to do it a different way with my other gem:

https://github.com/brendon/positioning/blob/main/.github/workflows/main.yml
https://github.com/brendon/positioning/blob/main/Gemfile

It's then easy to set up a matrix of tests across Rails, Ruby, and database libraries and only pick the compatible ones.

The downside is that this doesn't run well locally. If you want to try this way, by all means, but I realise it's quite an extra job in and of itself :) If the compile line above gets things working for you then just carry on I think :)

Reply:
I think it might be time to drop support for older rails/ruby versions. Did you first want me to have a go at getting rid of the appraisals gem?

Reply:
I think that would be great. I tried that too but I failed, since I don't know very well how that should be set up.

Reply:
That's all good. It's fairly complicated. I've done those changes on `master` not and the tests pass. It just tests Rails 6.1+ and Ruby 3.0+:

https://github.com/brendon/acts_as_list/actions/runs/9310848684

Feel free to update from master and then try out the tests. You can control the Rails, Ruby and DB versions via the command line:

RAILS_VERSION=7.1 (for example)
DB=sqlite | mysql | postgresql
Ruby: You just need to use rbenv or something like that and change your `.ruby-version` file.

Appraisals was good but a lot of work to maintain and people also used to not understand it and just edit the automatically generated bundler files :D

Reply:
You'll need to run bundle for each rails version you want to use:

`RAILS_VERSION=7.1 bundle`
`RAILS_VERSION=7.1 DB=postgresql bundle exec rake` (to run the tests)

Reply:
Perfect! That worked well. I was able to successfully run the tests both locally and through the CI on my fork.